wgpu-rs/src/lib.rs

```diff
@@ -233,6 +233,29 @@ impl<'a> TextureCopyView<'a> {
     }
 }
 
+pub struct CreateBufferMapped<'a, T> {
+    id: wgn::BufferId,
+    data: &'a mut [T],
+}
+
+impl<'a, T> CreateBufferMapped<'a, T>
+        where T: Copy {
+    pub fn fill_from_slice(self, slice: &[T]) -> Buffer {
+        self.data.copy_from_slice(slice);
+        self.finish()
+    }
+
+    pub fn as_slice_mut(&mut self) -> & mut [T] {
```

Comment:
is there a specific reason to not just expose `data` in public?

Reply:
Probably not. I'm generally wary of having an exposed data member, but we probably aren't at a point of API stability where the extra potential for breaking changes in the future is a problem.

Reply:
Might be a matter of taste. I find the `pub` just simpler to grasp than an opaque method, and I use `pub` whenever there is nothing bad the user could do with it.